### PR TITLE
8277814: ConcurrentRefineThread should report rate when deactivating

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change to the logging by concurrent refinement threads.
This change adds the cards processed per ms to the deactivation log message.

Testing:
mach5 tier1
locally ran a test with gc+refine=debug logging enabled and verified the
expected change to the deactivation log message.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277814](https://bugs.openjdk.java.net/browse/JDK-8277814): ConcurrentRefineThread should report rate when deactivating

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to b623710cf84dd184c3dabb2a56b57cdb310a92ea
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**) ⚠️ Review applies to b623710cf84dd184c3dabb2a56b57cdb310a92ea

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6550/head:pull/6550` \
`$ git checkout pull/6550`

Update a local copy of the PR: \
`$ git checkout pull/6550` \
`$ git pull https://git.openjdk.java.net/jdk pull/6550/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6550`

View PR using the GUI difftool: \
`$ git pr show -t 6550`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6550.diff">https://git.openjdk.java.net/jdk/pull/6550.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1ConcurrentRefineStats.cpp [hunk 1]
```diff
@@ -32,6 +32,12 @@ G1ConcurrentRefineStats::G1ConcurrentRefineStats() :
   _dirtied_cards(0)
 {}
 
+double G1ConcurrentRefineStats::refinement_rate_ms() const {
+  // Report 0 when no time recorded because no refinement performed.
+  double secs = refinement_time().seconds();
+  return (secs > 0) ? (refined_cards() / (secs * MILLIUNITS)) : 0.0;
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1ConcurrentRefineStats.cpp:38]: Not sure how this is used, but in similar calculations we only use and return a value if the elapsed time is large enough (something like a microsecond or so) to remove extreme outliers. Not sure if that is needed here too.

Comment (kimbarrett) [reply src/hotspot/share/gc/g1/g1ConcurrentRefineStats.cpp:38]: This change only uses it for logging.  I could just drop this function and do the equivalent calculation at the point where it's used.  I don't think a made up lower limit really makes that much sense anyway; I remember some places like you describe, but I thought many had been eliminated.  I could only find a couple.  The expectation here is that refined_cards() is probably also zero, unless the time granularity is rather coarse.

**Review by tschatzl — APPROVED** [review]

**Review by kstefanj — APPROVED** [review]
Looks good.

**Comment by kimbarrett** [comment]
Thanks @tschatzl and @kstefanj for reviews.

**Pull request closed without merging**